docs/tutorials/mobile/superres.md

```diff
@@ -0,0 +1,298 @@
+
+Create a file called MainActivity.kt and add the following pieces of code to it.
+
+1. Add import statements
+
+   ```kotlin
+   import ai.onnxruntime.*
+   import ai.onnxruntime.extensions.OrtxPackage
+   import android.annotation.SuppressLint
+   import android.os.Bundle
+   import android.widget.Button
+   import android.widget.ImageView
+   import android.widget.Toast
+   import androidx.activity.*
+   import androidx.appcompat.app.AppCompatActivity
+   import kotlinx.android.synthetic.main.activity_main.*
+   import kotlinx.coroutines.*
+   import java.io.InputStream
+   import java.util.*
+   import java.util.concurrent.ExecutorService
+   import java.util.concurrent.Executors
+   ```
+
+2. Add the class variables for the main activity class
+
+   ```kotlin
+   private var ortEnv: OrtEnvironment = OrtEnvironment.getEnvironment()
+   private var outputImage: ImageView? = null
+   private var superResolutionButton: Button? = null
+   ```
+
+3. Add the onCreate method
+
+   ```kotlin
+   override fun onCreate(savedInstanceState: Bundle?) {
+       super.onCreate(savedInstanceState)
+       setContentView(R.layout.activity_main)
+
+       outputImage = findViewById(R.id.imageView2);
+       superResolutionButton = findViewById(R.id.super_resolution_button)
+
+       superResolutionButton?.setOnClickListener {
+           performSuperResolution()
+               Toast.makeText(baseContext, "Super resolution performed!", Toast.LENGTH_SHORT).show()
+        }
+    }
+   ```
+
+4. Add the onDestroy method
+
+   ```kotlin
+   override fun onDestroy() {
+       super.onDestroy()
+       ortEnv.close()
+   }
+   ```
+
+5. Add the updateUI method
+
+   ```kotlin
+   private fun updateUI(result: Result) {
+   outputImage?.setImageBitmap(result.outputBitmap)
+   }
+   ```
+
+6. Add the readModel method
+
+   This method reads the ORT format model from the resources folder.
+
+   ```kotlin
+   private fun readModel(): ByteArray {
+       val modelID = R.raw.pt_super_resolution_op16
+       return resources.openRawResource(modelID).readBytes()
+   }   
+   ```
+
+7. Add the readInputImage method
+
+   This method reads a test image from the assets folder. Currently it reads a fixed image built into the application. The sample will soon be extended to read the image directly from the camera or the camera roll.
+
+   ```kotlin
+   private fun readInputImage(): InputStream {
+       return assets.open("test_superresolution.png")
+   }   
+   ```
+
+8. Add a method to create the ONNX Runtime session
+
+   A session holds a reference to the model used to perform inference in the application. It also takes a session options parameter, which is where you can specify different execution providers (hardware accelerators such as NNAPI). In this case, we default to running on CPU. We do however register the custom op library where the image encoding and decoding operators at the input and output of the model are found.
+
+   ```kotlin
```

Comment:
fyi, here I updated it into app onCreate() stage, and seems like not necessary to have a separate method for creating ort session anymore.

so probably can update to `Add steps to create the ONNX Runtime session`? Or let me know if it would be clearer to still keep a separate method for createSession()

```
override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        outputImage = findViewById(R.id.imageView2);
        superResolutionButton = findViewById(R.id.super_resolution_button)

        val sessionOptions: OrtSession.SessionOptions = OrtSession.SessionOptions()
        sessionOptions.registerCustomOpLibrary(OrtxPackage.getLibraryPath())

        superResolutionButton?.setOnClickListener {
            performSuperResolution(ortEnv.createSession(readModel(), sessionOptions))
            Toast.makeText(baseContext, "Super resolution performed!", Toast.LENGTH_SHORT).show()
        }
    }
```